app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt

```diff
@@ -0,0 +1,119 @@
+package com.mapbox.maps.testapp.utils
+
+import android.os.Handler
+import android.os.Looper
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.*
+
+class StyleSwitcher(
+  val mapboxMap: MapboxMap,
+  initialStyle: Style
+) {
+  private lateinit var currentStyle: Style
+  private val mainHandler = Handler(Looper.getMainLooper())
+  private val originalProperties = mutableMapOf<Pair<String, String>, Value>()
+  private var styleTransitionDelayMs: Long = 0
+  private var onStyleSwitched: ((Style) -> Unit)? = null
+
+  init {
+    mapboxMap.subscribe(
+      object : Observer() {
+        override fun notify(event: Event) {
+          when (event.type) {
+            // when map loaded, fade-in layers using original values
+            MapEvents.MAP_LOADED -> {
+              for (originalProperty in originalProperties) {
+                currentStyle.setStyleLayerProperty(
+                  originalProperty.key.first,
```

Comment:
can we assume the layer still exist?

Reply:
Are there any objections why it should not exist? IMHO `MAP_LOADED` is called almost immediately after `STYLE_LOADED` so those properties should be in place.

Reply:
But will add a check.

Reply:
I mean users could modify the style after you constructed the StyleSwitcher, the previously added layer/properties may not exist any more.